src/main.rs

```diff
@@ -117,35 +136,78 @@ fn main() {
     let command = if !matches.free.is_empty() {
         matches.free[0].clone()
     } else {
-        println!("{}", opts.usage("\n\tcobalt build"));
+        print_usage(opts);
         return;
     };
 
+    // Check for port and set port variable to it
+    let port = matches.opt_str("port").unwrap_or("3000".to_owned());
+
     match command.as_ref() {
         "build" => {
-            info!("Building from {} into {}", config.source, config.dest);
-            match cobalt::build(&config) {
-                Ok(_) => info!("Build successful"),
-                Err(e) => {
-                    error!("{}", e);
-                    error!("Build not successful");
-                    std::process::exit(1);
-                }
-            }
+            build(&config);
         }
 
         "serve" => {
-            info!("Serving {} through static file server", config.dest);
-            let mut server = Nickel::new();
-
-            server.utilize(StaticFilesHandler::new(&config.dest));
+            build(&config);
+            serve(&config.dest, &port);
+        }
 
-            server.listen("127.0.0.1:3000");
+        "watch" => {
+            build(&config);
+
+            let dest = config.dest.clone();
+            thread::spawn(move || {
+                serve(&dest, &port);
+            });
+
+            let (tx, rx) = channel();
+            let w: Result<RecommendedWatcher, Error> = Watcher::new(tx);
+
+            match w {
+                Ok(mut watcher) => {
+                    // TODO: clean up this unwrap
+                    watcher.watch(&config.source).unwrap();
+                    info!("watching {:?}", &config.source);
+
+                    loop {
+                        match rx.recv() {
+                            _ => {
+                                info!("Rebuilding cobalt site...");
+                                build(&config);
+                            }
+                        }
+                    }
+                }
+                Err(e) => error!("[Notify Error]: {}", e),
+            }
         }
 
         _ => {
-            println!("{}", opts.usage("\n\tcobalt build"));
+            print_usage(opts);
             return;
         }
     }
 }
+
+fn build(config: &Config) {
+    info!("Building from {} into {}", config.source, config.dest);
+    match cobalt::build(&config) {
+        Ok(_) => info!("Build successful"),
+        Err(e) => {
+            error!("{}", e);
+            error!("Build not successful");
+            std::process::exit(1);
+        }
+    };
+}
+
+fn serve(dest: &str, port: &str) {
+    info!("Serving {} through static file server", dest);
```

Comment:
maybe we should mention the port that it's being served on. I think you can also remove the "through static file server"